passkit_ui/lib/src/extensions/pk_pass_image_extension.dart

```diff
@@ -0,0 +1,10 @@
+import 'dart:typed_data';
+
+import 'package:passkit/passkit.dart';
+
+extension PkPassImageX on PkPassImage {
+  Uint8List forCorrectPixelRatio(double devicePixelRatio) {
+    // It just looks better when using images a step higher.
+    return fromMultiplier(devicePixelRatio.toInt() + 1);
+  }
+}
```

Comment:
While I understand your reason for changing this, I think it's more important to make it easy to use here, especially now that you made it public facing. After all, that's what libraries for, to make the life of the consumer of the library easier. Also, a pass has at max like 3 or 4 images and one or three calls to the MediaQuery isn't really anything that makes or breaks the performance. 

By keeping the `BuildContext` as parameter, we can also change how the correct image is retrieved later ta a different property of the MediaQuery instead, so it future-proofes the API in a way that doesn't break consumers.
Keeping the `BuildContext` as a parameter also makes the API more Flutter-like, since Flutter also always just takes a context as a parameter.

If it was just an internal API I wouldn't mind this change at all though.

Reply:
BuildContext shouldn't be a parameter for external methods, it should be used exclusively on the widgets' build method to avoid passing the whole context object as a function parameter and also causing context issues. It's even easier to test methods by passing a devicePixelRatio in this case than a mock build context. I think breaking changes are inevitable on packages and in this case it's a small one so it shouldn't be an issue!